Firebase/Auth/Source/SystemService/FIRAuthNotificationManager.m

```diff
@@ -107,16 +118,25 @@ - (void)checkNotificationForwardingWithCallback:(FIRAuthNotificationForwardingCa
         kNotificationProberKey : @"This fake notification should be forwarded to Firebase Auth."
       }
     };
+#if !TARGET_OS_WATCH
     if ([self->_application.delegate respondsToSelector:
             @selector(application:didReceiveRemoteNotification:fetchCompletionHandler:)]) {
       [self->_application.delegate application:self->_application
             didReceiveRemoteNotification:proberNotification
                   fetchCompletionHandler:^(UIBackgroundFetchResult result) {}];
-#if !TARGET_OS_TV
+#endif
+#if !TARGET_OS_TV && !TARGET_OS_WATCH
     } else if ([self->_application.delegate respondsToSelector:
                    @selector(application:didReceiveRemoteNotification:)]) {
       [self->_application.delegate application:self->_application
             didReceiveRemoteNotification:proberNotification];
+#endif
+#if TARGET_OS_WATCH
+    if ([self->_application.delegate respondsToSelector:
+            @selector(didReceiveRemoteNotification:fetchCompletionHandler:)]) {
+      [self->_application.delegate
+            didReceiveRemoteNotification:proberNotification
+                  fetchCompletionHandler:^(WKBackgroundFetchResult result) {}];
 #endif
```

Comment:
I think this could be cleaned up a little bit:

```
#if TARGET_OS_WATCH
  // lines 135-139,  `didReceiveRemoteNotification:fetchCompletionHandler:`
#else 
  // lines 122-126, `application:didReceiveRemoteNotification:fetchCompletionHandler:`
#endif 

#if !TARGET_OS_TV && !TARGET_OS_WATCH
 // Lines 129 to 132
#endif
```

I think that should work? It may take a bit of playing around with the closing curly braces. Let me know if I'm wrong though and we can revisit!